### PR TITLE
ToDo List Milestone 2: Add Features: Add and Remove

### PR DESCRIPTION
**Pull request by AyoubM490**
In this Milestone, I:
- Added the Add and Remove features
- used ES6

### CODE REVIEW

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @AyoubMs 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct Github Flow.
✅ Implemented a function for adding a new task
✅ Implemented a function for editing task descriptions
✅ Implemented a function for deleting a task

## Required Changes ♻️

- [x] All changes to the To Do List should be saved in local storage.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/storage.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+};
+
+const getStorage = () => {
+  const tasks = localStorage.getItem('todo-list')
+    ? JSON.parse(localStorage.getItem('todo-list'))
+    : [];
+  return tasks;
+};
+
+export { saveStorage, getStorage };
```
Comment (FrederickMih) [inline src/modules/storage.js:12]: - [x] Please let's implement the changes requested by the previous reviewer. Tasks added to the list are not still saved in the local storage.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AyoubMs , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

- [OPTIONAL] Please also implement a function for editing task descriptions 

- [x] Some of your linters are also failing. Please fix it.


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/storage.js (hunk 1 above)
Comment (AyoubM490) [reply src/modules/storage.js:12]: Done

File: src/modules/addTask.js [hunk 2]
```diff
@@ -24,22 +30,29 @@ const addTask = (task) => {
...
     }
   };
 
+  const remove = () => {
+    container.removeChild(li);
+    removeTask(li, tasks);
+  };
   input.addEventListener('click', inputChecked);
+  span.addEventListener('click', () => {
+    remove();
```
Comment (dasileker) [inline src/modules/addTask.js:54]: - [x] As part of the requirement the const remove should take the **clear all completed button** (if the check-box = true => remove item from the array), but **the clear all completed btn** is not working and the **trash icon** it remove everything.

**Review by dasileker — CHANGES_REQUESTED** [review]
## **Hi @AyoubMs ,** 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

- Please make sure that thrash icon remove just the targeted object from the array and the clear all complete remove all remove everything checked in the array.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/addTask.js (hunk 2 above)
Comment (AyoubM490) [reply src/modules/addTask.js:54]: I think I deployed the wrong version, please don't look for the live version for now

Comment (AyoubM490) [reply src/modules/addTask.js:54]: The Clear all button is intended for the milestone 3 (next milestone), please don't invalidate me for that reason

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @AyoubMs, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 
- Great job implementing the changes requested


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**